extensions/wikia/AdMixExperiment/scripts/noVideo1.js

```diff
@@ -0,0 +1,269 @@
+		var visibleElementBeforeWrapperHeight;
+
+		// TODO that should be set to true for ad mix variant XW-3156
+		var topRightAdFixed = false;
+
+		function getAdAndRecircWrapperTopPosition() {
+			var visibleElementBeforeWrapperBottom = parseInt($visibleElementBeforeWrapper.offset().top, 10) +
+				visibleElementBeforeWrapperHeight;
+
+			if (recircEnabled) {
+				return visibleElementBeforeWrapperBottom - globalNavigationHeight;
+			} else {
+				return visibleElementBeforeWrapperBottom + noRecircAdBottomPosition -
+					(viewportHeight - $adAndRecircWrapper.outerHeight(true));
+			}
+		}
+
+		function getAdAndRecircWrapperBottomPosition(startPosition) {
+			return startPosition + gapSize;
+		}
+
+		function getFirstAdTopPosition() {
+			return $rail.offset().top - globalNavigationHeight;
+		}
+
+		function getFirstAdBottomPosition() {
+			return getFirstAdTopPosition() + gapSize;
+		}
+
+		function getStopPosition() {
+			var stopPoint = parseInt($footer.offset().top, 10);
+			adMixRecircWrapperHeight = $adAndRecircWrapper.outerHeight(true);
+
+			return stopPoint - globalNavigationHeight - adMixRecircWrapperHeight - bottomMargin;
+		}
+
+		function resetRecircStyles() {
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', gapSize + 'px');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+			}
+
+			$adAndRecircWrapper.css({
+				bottom: '',
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function reset() {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+			resetRecircStyles();
+		}
+
+		function scrollAfterAdAndRecircWrapperTopPosition() {
+			var adAndRecircWrapperStyles = {
+				position: '',
+				width: ''
+			};
+
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', gapSize + 'px');
+
+				$recirc.css({
+					position: 'fixed',
+					top: globalNavigationHeight + 'px',
+					width: '280px'
+				});
+
+				adAndRecircWrapperStyles.top = '';
+			} else {
+				adAndRecircWrapperStyles.bottom = '';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAfterAdAndRecircWrapperBottomPosition() {
+			var adAndRecircWrapperStyles = {
+				position: 'fixed',
+				width: '300px'
+			};
+
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', '0');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+
+				adAndRecircWrapperStyles.top = globalNavigationHeight + 'px';
+			} else {
+				adAndRecircWrapperStyles.bottom = noRecircAdBottomPosition + 'px';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAtBottom(stopPosition) {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			// FIXME cache it? at least the element
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)